src/buildCommon.js

```diff
@@ -308,30 +368,50 @@ const makeVList = function({positionType, positionData, children}, options) {
 
             currPos = currPos + diff;
 
-            children.push({type: "kern", size: size});
+            children.push({type: "kern", size});
             children.push(oldChildren[i]);
         }
-    } else if (positionType === "top") {
+
+        return {children, depth};
+    }
+
+    let depth;
+    if (params.positionType === "top") {
         // We always start at the bottom, so calculate the bottom by adding up
         // all the sizes
-        let bottom = positionData;
-        for (i = 0; i < children.length; i++) {
-            if (children[i].type === "kern") {
-                bottom -= children[i].size;
-            } else {
-                bottom -= children[i].elem.height + children[i].elem.depth;
-            }
+        let bottom = params.positionData;
+        for (const child of params.children) {
+            bottom -= child.type === "kern" ?
+                child.size :
+                child.elem.height + child.elem.depth;
```

Comment:
Style nit, if a ternary is too long to fit on a single line can you format it as:
```
result = condition
   ? truthyResult
   : falsyResult;
```
I find it makes it easier to tell that it's a ternary.